native/components/message-markdown-rules.react.js

```diff
@@ -0,0 +1,155 @@
+};
+
+type ReactElement = React$Element<any>;
+
+type Output<Result> = (node: ASTNode, state?: ?State) => Result;
+
+export default function(styles: any, opts: any = {}) {
+  const LINK_INSIDE = '(?:\\[[^\\]]*\\]|[^\\]]|\\](?=[^\\[]*\\]))*';
+  const LINK_HREF_AND_TITLE =
+    '\\s*<?([^\\s]*?)>?(?:\\s+[\'"]([\\s\\S]*?)[\'"])?\\s*';
+  var pressHandler = function(target) {
+    if (opts.onLink) {
+      opts.onLink(target).catch(function(error) {
+        console.log(
+          'There has been a problem with this action. ' + error.message,
+        );
+        throw error;
+      });
+    }
+  };
+  return {
+    autolink: {
+      react: function(
+        node: SingleASTNode,
+        output: Output<ReactElement>,
+        state: State,
+      ) {
+        state.withinText = true;
+        const _pressHandler = () => {
+          pressHandler(node.target);
+        };
+        const autolink = React.createElement(
+          Text,
+          {
+            key: state.key,
+            style: styles.autolink,
+            onPress: _pressHandler,
+          },
+          output(node.content, state),
+        );
+        return autolink;
+      },
+    },
+    link: {
+      match: SimpleMarkdown.inlineRegex(
+        new RegExp(
+          '^\\[(' + LINK_INSIDE + ')\\]\\(' + LINK_HREF_AND_TITLE + '\\)',
+        ),
+      ),
+      react: function(
+        node: SingleASTNode,
+        output: Output<ReactElement>,
+        state: State,
+      ) {
+        state.withinLink = true;
+        const _pressHandler = () => {
+          pressHandler(node.target);
+        };
+        const link = React.createElement(
+          Text,
+          {
+            key: state.key,
+            style: styles.autolink,
+            onPress: _pressHandler,
+          },
+          output(node.content, state),
+        );
+        state.withinLink = false;
+        return link;
+      },
+    },
+    paragraph: {
+      react: function(
+        node: SingleASTNode,
+        output: Output<ReactElement>,
+        state: State,
+      ) {
+        const paragraph = React.createElement(
+          Text,
+          {
+            key: state.key,
+            style: styles.paragraph,
+          },
+          output(node.content, state),
+        );
+        return paragraph;
+      },
+    },
+    text: {
+      react: function(
+        node: SingleASTNode,
+        output: Output<string>,
+        state: State,
+      ) {
+        let textStyle = {
+          ...styles.text,
+          ...(state.style || {}),
+        };
+
+        if (state.withinLink) {
+          textStyle = [styles.text, styles.autolink];
+        }
+
+        if (state.withinQuote) {
+          textStyle = [styles.text, styles.blockQuoteText];
+        }
+
+        const text = React.createElement(
+          Text,
+          {
+            key: state.key,
+            style: textStyle,
+          },
+          node.content,
+        );
+        return text;
+      },
+    },
+    url: {
```

Comment:
If `url` is just the `<https://google.com>` syntax, let's get rid of it for now. We're only trying to tackle `autolink` in this PR. The goal is for the behavior to be the same as before

Reply:
actually, I think simple-markdown uses `url` for matching `https`, `autolink` regex is for `<https://google.com>` syntax, among others